[Raman Gupta] iP

eyelessrhyme7 opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
[Google](https://google.com)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
* easy to learn
+ ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 👍 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Use `git push` to push. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

eyelessrhyme7 committed:
Completed Level-0

eyelessrhyme7 committed:
Completed Level-1

eyelessrhyme7 committed:
Completed Level-2

eyelessrhyme7 committed:
Completed Level-3

eyelessrhyme7 committed:
Completed Level-4

eyelessrhyme7 committed:
Completed A-TextUiTesting

eyelessrhyme7 committed:
Completed Level-5

eyelessrhyme7 committed:
Completed Level-6

eyelessrhyme7 committed:
Completed Level-7

eyelessrhyme7 committed:
Merge branch 'branch-Level-7'

Elijah5399 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+public class Deadline extends Task {
+
+    protected String by;
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public static Deadline DeadlineCon(String description, String by) throws InvalidTaskCreationException {
+        if (description.equalsIgnoreCase("")) {
+            throw new InvalidTaskCreationException("OOPS!!! The description of a Deadline Task cannot be empty.");
+        } else if (by.equalsIgnoreCase("")) {
+            throw new InvalidTaskCreationException("OOPS!!! The deadline time of a Deadline Task cannot be empty.");
+        } else {
+            return new Deadline(description, by);
+        }
+    }
```
It's great that the DeadlineCon method is relatively small. However, would you like to consider using JavaDoc to explain what the code does? Also, "DeadlineCon" is a method and should be in camel case. You may wish to rename the "Con" term to a full word.

Elijah5399 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,227 @@
+    }
+
+    public static final String greeting = "Hello, I'm Capt. Price! What can I do for you today, Sergeant?";
+
+    public static final String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    public static final String horizontalBar = "-------------------------------------------------";
+    public static final String exitGreeting = "Over and out. See you next mission!";
```
These are some interesting greetings! :+1: Would you like to consider making the constant names all caps?

Elijah5399 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+        this.isDone = false;
+    }
+
+    public static Task taskCon(String userInput) throws InvalidCommandException, InvalidTaskCreationException{
```
Would you like to rename this variable's 'con' term to an actual word, and include JavaDoc? I'm sure it will be helpful to readers!

Elijah5399 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,227 @@
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.sql.SQLOutput;
```
Is this import statement necessary? :thinking:

eyelessrhyme7 committed:
Completed Level-8

eyelessrhyme7 committed:
Merge branch 'branch-Level-8'

eyelessrhyme7 committed:
Completed A-MoreOOP

eyelessrhyme7 committed:
Completed A-Packages

eyelessrhyme7 committed:
Merge branch 'add-gradle-support' of https://github.com/eyelessrhyme7/ip into add-gradle-support

eyelessrhyme7 committed:
Merge branch 'add-gradle-support'

eyelessrhyme7 committed:
Completed A-JUnit

eyelessrhyme7 committed:
Completed A-CodingStandard

eyelessrhyme7 committed:
Completed A-JavaDoc

eyelessrhyme7 committed:
Completed Level-9

eyelessrhyme7 committed:
Merge branch 'branch-Level-9'

eyelessrhyme7 committed:
Merge branch 'branch-A-JavaDoc'

eyelessrhyme7 committed:
Merge branch 'branch-A-CodingStandard'

Bryan-Goh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+ * The `TaskList` class manages a list of tasks in the Duke application.
+ */
+public class TaskList {
+    private ArrayList<Task> taskArrayList;
```
Perhaps you could use a better variable name?

Bryan-Goh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+     *
+     * @return A TaskList containing tasks due within one week.
+     */
+    public TaskList dueWithinWeek() {
```
Perhaps you can use a verb for the method?

Bryan-Goh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        while ((line = reader.readLine()) != null) {
+            switch (line.charAt(0)) {
```
I am not sure I like how the switch case method have the wrong indentation. Perhaps you could improve it?

Bryan-Goh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,74 @@
+     * @param message The message to format.
+     * @return The formatted message.
+     */
+    private String botMessage(String message) {
```
Perhaps this method can have a better name?

Bryan-Goh reviewed (commented):
Looks Good To Me. Some naming and indentation change recommended

eyelessrhyme7 committed:
Completed A-CheckStyle

eyelessrhyme7 committed:
Merge tag 'A-CheckStyle' into branch-A-CheckStyle

A-CheckStyle

eyelessrhyme7 committed:
Merge branch 'branch-A-CheckStyle'

eyelessrhyme7 committed:
Completed Level-10

eyelessrhyme7 committed:
Merge branch 'branch-Level-10'

eyelessrhyme7 committed:
Completed A-Varargs

eyelessrhyme7 committed:
Merge branch 'branch-A-Varargs'

eyelessrhyme7 committed:
Complete New Jar File Creation and Fix UI Padding Issue

eyelessrhyme7 committed:
Complete Jar File Creation and Fix UI Padding Issue

Complete Jar File Creation

Previously, our jar files were CLI-based, which limited the usability
and user-friendliness of our application. In this commit, we've
successfully transitioned to a JavaFX-based GUI, enhancing the user
experience and accessibility of our software. This change makes our
application more intuitive and visually appealing.

Fix UI Padding Issue

One of the notable UI issues we addressed was related to the dialog
boxes. Previously, these dialog boxes lacked proper padding, resulting
in a cramped and unappealing appearance. With this commit, we've
implemented consistent and aesthetically pleasing padding in all
dialog boxes, ensuring a more polished and comfortable user interface.

These enhancements mark a significant improvement in the overall
usability and visual appeal of our application, making it more
user-friendly and visually engaging.

eyelessrhyme7 committed:
Completed A-Assertions

eyelessrhyme7 committed:
Merge pull request #1 from eyelessrhyme7/branch-A-Assertions

Completed A-Assertions

eyelessrhyme7 committed:
Completed A-CodeQuality

eyelessrhyme7 committed:
Merge pull request #2 from eyelessrhyme7/branch-A-CodeQuality

Completed A-CodeQuality

eyelessrhyme7 committed:
Completed A-Streams

eyelessrhyme7 committed:
Merge pull request #3 from eyelessrhyme7/branch-A-Streams

Completed A-Streams

eyelessrhyme7 committed:
Completed A-CI

eyelessrhyme7 committed:
BCD-Extension

B-DoAfterTasks

In this commit, we've added the ability to set task dependencies,
allowing you to establish parent-child relationships between tasks.
You can now define which tasks rely on the completion of others,
creating a more structured workflow management system.

C-Tagging

This commit introduces a tagging feature. You can now assign tags to
your tasks, such as #fun and #sport, for better organization. Tags
provide context and make it easier to categorize and filter tasks.

These enhancements improve the overall usability and flexibility of
the BCD-Extension, enhancing task management capabilities.

eyelessrhyme7 committed:
Merge pull request #4 from eyelessrhyme7/branch-BCD-Extension

BCD-Extension

eyelessrhyme7 committed:
Completed A-BetterGui

eyelessrhyme7 committed:
Merge pull request #6 from eyelessrhyme7/branch-A-BetterGui

Completed A-BetterGui

eyelessrhyme7 committed:
Completed A-Personality

eyelessrhyme7 committed:
Merge pull request #7 from eyelessrhyme7/branch-A-Personality

Completed A-Personality

eyelessrhyme7 committed:
Completed A-MoreTesting

eyelessrhyme7 committed:
Merge branch 'branch-A-MoreTesting'

eyelessrhyme7 committed:
Complete A-MoreErrorHandling

eyelessrhyme7 committed:
Merge branch 'branch-A-MoreErrorHandling'

eyelessrhyme7 committed:
Complete A-Image

eyelessrhyme7 committed:
Complete A-UserGuide

eyelessrhyme7 committed:
Merge branch 'branch-A-UserGuide'

eyelessrhyme7 committed:
Complete A-UserGuide

eyelessrhyme7 committed:
Update README.md

eyelessrhyme7 committed:
Complete A-Ui

eyelessrhyme7 committed:
Merge branch 'master' of https://github.com/eyelessrhyme7/ip

eyelessrhyme7 committed:
Update README.md

eyelessrhyme7 committed:
Update README.md

eyelessrhyme7 committed:
Update README.md

eyelessrhyme7 committed:
Update README.md
